app/controllers/application_controller/ci_processing.rb

```diff
@@ -637,14 +649,21 @@ def generic_button_operation(action, action_name, operation, options = {})
   def testable_action(action)
     controller = params[:controller]
     vm_infra_untestable_actions = %w[check_compliance_queue destroy refresh_ems vm_miq_request_new]
-    ems_cluster_untestable_actions = %w[scan]
-    if controller == "vm_infra"
-      return vm_infra_untestable_actions.exclude?(action)
-    end
-    if controller == "ems_cluster"
-      return ems_cluster_untestable_actions.exclude?(action)
+    ems_cluster_untestable_actions = %w[check_compliance_queue scan]
+    other_untestable_actions = %w[check_compliance_queue]
+
+    return false if @display == 'ems_clusters' && action == 'scan'
+
+    case controller
+    when 'ems_cluster'
+      ems_cluster_untestable_actions.exclude?(action)
+    when 'auth_key_pair_cloud', 'availability_zone', 'cloud_network', 'cloud_tenant', 'ems_cloud', 'ems_infra', 'flavor', 'host', 'host_aggregate', 'resource_pool', 'storage', 'vm_cloud'
+      other_untestable_actions.exclude?(action)
+    when 'vm_infra'
```

Comment:
Just checking..

`destroy`, `refresh_ems` and `vm_miq_request_new` are OK for `vm_cloud` but not `vm_infra`?

Reply:
> Just checking..
> 
> `destroy`, `refresh_ems` and `vm_miq_request_new` are OK for `vm_cloud` but not `vm_infra`?

these changes are just a backport of an upstream PR.

Reply:
Ah, good point :+1: